parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -48,22 +48,22 @@ class Manager:
                 |                          |                IPC-Qeueues
 
     """
-    def __init__(self,
-                 addresses="127.0.0.1",
-                 address_probe_timeout=30,
-                 task_port="50097",
-                 result_port="50098",
-                 cores_per_worker=1,
-                 mem_per_worker=None,
-                 max_workers=float('inf'),
-                 prefetch_capacity=0,
-                 uid=None,
-                 block_id=None,
-                 heartbeat_threshold=120,
-                 heartbeat_period=30,
-                 poll_period=10,
-                 cpu_affinity=False,
-                 available_accelerators: Sequence[str] = ()):
+    def __init__(self, *,
+                 addresses,
+                 address_probe_timeout,
+                 task_port,
+                 result_port,
+                 cores_per_worker,
+                 mem_per_worker,
+                 max_workers,
+                 prefetch_capacity,
+                 uid,
+                 block_id,
+                 heartbeat_threshold,
+                 heartbeat_period,
+                 poll_period,
+                 cpu_affinity,
+                 available_accelerators: Sequence[str]):
```

Comment:
We may want to set defaults in the argparser for these to maintain consistency with the other parameters and avoid breaking any workflows that call `process_worker_pool.py` directly.

Reply:
I don't think removing defaults in this thing breaks any invocations that work prior to this PR: for example, Manager had a task port default of `"50097"` - but default that isn't accessible: the arg parser doesn't let process_worker_pool even get that far.

```
$ process_worker_pool.py 
usage: process_worker_pool.py [-h] [-d] [-a ADDRESSES] [-l LOGDIR] [-u UID] [-b BLOCK_ID] [-c CORES_PER_WORKER]
                              [-m MEM_PER_WORKER] -t TASK_PORT [--max_workers MAX_WORKERS] [-p PREFETCH_CAPACITY]
                              [--hb_period HB_PERIOD] [--hb_threshold HB_THRESHOLD]
                              [--address_probe_timeout ADDRESS_PROBE_TIMEOUT] [--poll POLL] -r RESULT_PORT
                              [--cpu-affinity {none,block,alternating,block-reverse}]
                              [--available-accelerators [AVAILABLE_ACCELERATORS ...]]
process_worker_pool.py: error: the following arguments are required: -t/--task_port, -r/--result_port
```

I'd really like to, in a different PR, remove more defaults from the arg parser. 

At least one default is broken there:

```
$ process_worker_pool.py -t 10000 -r 10001
Traceback (most recent call last):
  File "/home/benc/parsl/virtualenv-3.11/bin/process_worker_pool.py", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/benc/parsl/src/parsl/parsl/executors/high_throughput/process_worker_pool.py", line 765, in <module>
    logger.info("Accelerators: {}".format(" ".join(args.available_accelerators)))
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: can only join an iterable
```

and I think the cpu_affinity default of `None` will break things if actually used (although I haven't tested that).

Some of the defaults this PR removes from Manager don't make sense as defaults: for example, default port numbers for task and result connections, which need to be coordinated with the corresponding interchange rather than defaulted.

Reply:
That works for me. :rocket: